File: backend/app/schemas/stock.py
from datetime import date

from pydantic import BaseModel, ConfigDict


class StockUpdate(BaseModel):
    quantity: int | None = None
    low_stock_threshold: int | None = None
    price_override: float | None = None
    zone_id: int | None = None


class StockOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    product_id: int
    branch_id: int
    quantity: int
    low_stock_threshold: int
    price_override: float | None = None
    zone_id: int | None = None
    product_name: str
    sku: str
    best_before_date: date | None = None
    effective_price: float


class BranchStockOut(BaseModel):
    """Bir ürünün, çağıranın yetki alanındaki (kendi şube/bölge/şirket) her şubedeki durumu —
    PROCESS.md Faz 3 "quantity takibi" (2026-08-11). region_id/region_name sadece general_manager
    için doldurulur (2026-08-12 eklendi — company scope'ta şube listesini bölgeye göre gruplamak
    için); region_manager/branch_manager'da zaten tek bölge/şube olduğu için None kalır."""

    branch_id: int
    branch_name: str
    region_id: int | None = None
    region_name: str | None = None
    quantity: int
    low_stock_threshold: int
    effective_price: float

File: frontend/src/types/stock.ts
// backend/app/schemas/stock.py::StockOut ile birebir eşleşir.
export interface StockItem {
  product_id: number;
  branch_id: number;
  quantity: number;
  low_stock_threshold: number;
  price_override: number | null;
  zone_id: number | null;
  product_name: string;
  sku: string;
  best_before_date: string | null;
  effective_price: number;
}

// backend/app/schemas/stock.py::BranchStockOut ile birebir eşleşir (Faz 3 "quantity takibi").
// region_id/region_name sadece general_manager'da (company scope) dolu gelir.
export interface BranchStockItem {
  branch_id: number;
  branch_name: string;
  region_id: number | null;
  region_name: string | null;
  quantity: number;
  low_stock_threshold: number;
  effective_price: number;
}

export interface StockUpdatePayload {
  quantity?: number;
  low_stock_threshold?: number;
  price_override?: number | null;
  zone_id?: number | null;
}
